### PR TITLE
Add functions `unlines'`, `unlines` to Prelude.String

### PR DESCRIPTION
**Pull request by xged**
Solves #2464

I am leaving all of the necessary further-integration/followup-changes for more experienced ones :-)

### CODE REVIEW

File: libs/prelude/Prelude/Strings.idr [hunk 1]
```diff
@@ -276,6 +266,34 @@ unwords' ws = assert_total (foldr1 addSpace ws)
...
+||| Joins the character lists by newlines into a single character list.
+|||
+||| ```idris example
+||| unlines' [['l','i','n','e'], ['l','i','n','e','2'], ['l','n','3'], ['D']]
+||| ```
+unlines' : List (List Char) -> List Char
+unlines' [] = []
+unlines' ws = assert_total (foldr1 addLine ws) where
+  addLine : List Char -> List Char -> List Char
+  addLine w s = w ++ ('\n' :: s)
```
Comment (melted) [inline libs/prelude/Prelude/Strings.idr]: Just a nit, but `w` in unwords presumably stands for "word", so `l` would be more logical here.

**Comment by melted** [comment]
looks useful!

File: libs/prelude/Prelude/Strings.idr (hunk 1 above)
Comment (xged) [inline libs/prelude/Prelude/Strings.idr]: Yes it has to be `l` (silly me).

Comment (xged) [inline libs/prelude/Prelude/Strings.idr]: I am a *newb* so do all of the necessary corrections/integration for me :-)

**Comment by xged** [comment]
I am a newb so do all of the necessary corrections/integration for me :-)

**Comment by Melvar** [comment]
You seem to have moved `foldr1` to *after* it is used.

**Comment by xged** [comment]
My bad, I will create new PR.

**Pull request closed without merging**